src/strategies/polygon/strategy-general-masterchef-base.sol

```diff
@@ -0,0 +1,190 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.6.7;
+
+import "./strategy-base.sol";
+import "../../interfaces/masterchef.sol";
+
+abstract contract StrategyGeneralMasterChefBase is StrategyBase {
+    // Token addresses
+    address public masterchef;
+    address public rewardToken;
+
+    address public token0;
+    address public token1;
+
+    // How much TITAN tokens to keep?
+    uint256 public keepReward = 0;
+    uint256 public constant keepRewardMax = 10000;
+
+    uint256 public poolId;
+    mapping (address => address[]) public uniswapRoutes;
+
+    constructor(
+        address _rewardToken,
+        address _masterchef,
+        address _token0,
+        address _token1,
+        uint256 _poolId,
+        address _lp,
+        address _governance,
+        address _strategist,
+        address _controller,
+        address _timelock
+    )
+        public
+        StrategyBase(
+            _lp,
+            _governance,
+            _strategist,
+            _controller,
+            _timelock
+        )
+    {
+        poolId = _poolId;
+        token0 = _token0;
+        token1 = _token1;
+        rewardToken = _rewardToken;
+        masterchef = _masterchef;
+    }
+    
+    function balanceOfPool() public override view returns (uint256) {
+        (uint256 amount, ) = IMasterchef(masterchef).userInfo(poolId, address(this));
+        return amount;
+    }
+
+    function getHarvestable() external virtual view returns (uint256) {
+        uint256 _pendingReward = IMasterchef(masterchef).pendingReward(poolId, address(this));
+        return _pendingReward;
+    }
+
+    // **** Setters ****
+
+    function deposit() public override {
+        uint256 _want = IERC20(want).balanceOf(address(this));
+        if (_want > 0) {
+            IERC20(want).safeApprove(masterchef, 0);
+            IERC20(want).safeApprove(masterchef, _want);
+            IMasterchef(masterchef).deposit(poolId, _want);
+        }
+        _want = IERC20(want).balanceOf(address(this));
```

Comment:
Updating the `_want` balance appears unecessary

Reply:
removed it